src/InteractiveWindow/Editor/InteractiveWindow_UIThread.cs

```diff
@@ -531,9 +529,15 @@ public Task SubmitAsync()
 
                     StartCursorTimer();
 
-                    Debug.Assert(_currentTask == null, "Shouldn't be either executing or resetting");
-                    _currentTask = _window._evaluator.ExecuteCodeAsync(snapshotSpan.GetText());
-                    return _currentTask.ContinueWith(FinishExecute, _uiScheduler);
+                    var executionResult = await _window._evaluator.ExecuteCodeAsync(snapshotSpan.GetText()).ConfigureAwait(true);
+                    Debug.Assert(_window.OnUIThread()); // ConfigureAwait should bring us back to the UI thread.
+
+                    Debug.Assert(State == State.ExecutingInput || State == State.Resetting, $"Unexpected state {State}");
+                    
+                    if (State == State.ExecutingInput)
+                    {
+                        FinishExecute(executionResult.IsSuccessful);
```

Comment:
Will FinishExecute be executed if ExecuteCodeAsync throws a non-fatal exception?